php/src/Docker/DockerActionManager.php

```diff
@@ -35,98 +38,74 @@ private function BuildImageName(Container $container) : string {
-        } catch (RequestException $e) {
-            if ($e->getCode() === 404) {
-                return ContainerState::ImageDoesNotExist;
-            }
-            throw $e;
-        }
-
-        $responseBody = json_decode((string)$response->getBody(), true);
-
-        if ($responseBody['State']['Restarting'] === true) {
-            return ContainerState::Restarting;
-        } else {
-            return ContainerState::NotRestarting;
-        }
-    }
-
-    public function GetContainerUpdateState(Container $container) : VersionState
-    {
+    public function GetUpdateState(Container $container): UpdateState {
         $tag = $container->GetImageTag();
         if ($tag === '%AIO_CHANNEL%') {
             $tag = $this->GetCurrentChannel();
         }
 
         $runningDigests = $this->GetRepoDigestsOfContainer($container->GetIdentifier());
         if ($runningDigests === null) {
-            return VersionState::Different;
+            return UpdateState::Outdated;
         }
         $remoteDigest = $this->dockerHubManager->GetLatestDigestOfTag($container->GetContainerName(), $tag);
         if ($remoteDigest === null) {
-            return VersionState::Equal;
+            return UpdateState::Latest;
         }
 
-        foreach($runningDigests as $runningDigest) {
-            if ($runningDigest === $remoteDigest) {
-                return VersionState::Equal;
-            }
-        }
-        return VersionState::Different;
+        return in_array($remoteDigest, $runningDigests, true) ? UpdateState::Latest : UpdateState::Outdated;
     }
 
-    public function GetContainerStartingState(Container $container) : ContainerState
-    {
-        $runningState = $this->GetContainerRunningState($container);
-        if ($runningState === ContainerState::Stopped || $runningState === ContainerState::ImageDoesNotExist) {
-            return $runningState;
+    /** @throws GuzzleException */
+    public function GetContainerState(Container $container): ContainerState {
+        $url = $this->BuildApiUrl(sprintf('containers/%s/json', urlencode($container->GetIdentifier())));
+        try {
+            $response = $this->guzzleClient->get($url);
+        } catch (GuzzleException $e) {
+            if ($e->getCode() === 404) {
+                return ContainerState::DoesNotExist;
+            }
+            throw $e;
         }
 
-        $containerName = $container->GetIdentifier();
-        $internalPort = $container->GetInternalPort();
-        if($internalPort === '%APACHE_PORT%') {
-            $internalPort = $this->configurationManager->GetApachePort();
-        } elseif($internalPort === '%TALK_PORT%') {
-            $internalPort = $this->configurationManager->GetTalkPort();
+        $body = json_decode($response->getBody()->getContents(), true);
+        assert(is_array($body));
+        assert(is_array($body['State']));
+
+        $state = match ($body['State']['Status']) {
+            'running' => ContainerState::Healthy,
```

Comment:
For example this should be ContainerState::Running, no?

Reply:
See my comment